monkey/monkey_island/cc/app.py

```diff
@@ -109,75 +111,81 @@ def init_app_url_rules(app):
 
 
 class FlaskDIWrapper:
+    class URLAlreadyExistsError(Exception):
+        pass
+
     def __init__(self, api: flask_restful.Api, container: DIContainer):
         self._api = api
         self._container = container
 
-    def add_resource(self, resource: Type[flask_restful.Resource], *urls: str):
+    def add_resource(self, resource: Type[IResource]):
         dependencies = self._container.resolve_dependencies(resource)
-        self._api.add_resource(resource, *urls, resource_class_args=dependencies)
+        FlaskDIWrapper._check_for_duplicate_urls(self._api.app.url_map, resource.urls)
```

Comment:
We need to handle the exception here.

Reply:
Why? The `FlaskDIWrapper` shouldn't be responsible for deciding how to deal with this error.